vpr/src/route/sink_sampling.h

```diff
@@ -0,0 +1,155 @@
+#pragma once
+
+/** @file Sink downsamplers for parallel routing. 
+ *
+ * These are used to get a "minimal skeleton routing" from the main task. 
+ * Rest of the routing is delegated to child tasks. They will work with a
+ * strictly limited bounding box, so it's necessary that the initial routing
+ * provides enough hints while routing to as few sinks as possible to limit
+ * the serial bottleneck. */
+
+#include <cmath>
+#include <limits>
+#include <vector>
+#include "globals.h"
+#include "partition_tree.h"
+#include "route_common.h"
+#include "router_lookahead_sampling.h"
+
+/** Sink container for geometry operations */
+struct SinkPoint {
+    int x;
+    int y;
+    int isink;
+
+    bool operator==(const SinkPoint& rhs) const {
+        return x == rhs.x && y == rhs.y;
+    }
+    bool operator<(const SinkPoint& rhs) const {
+        if (x < rhs.x)
+            return true;
+        if (x > rhs.x)
+            return false;
+        if (y < rhs.y)
+            return true;
+        if (y > rhs.y)
+            return false;
+        return isink < rhs.isink;
+    }
+};
+
+/* Namespace the following to keep these generic fn names available */
+namespace sink_sampling {
+
+/* TODO: Replace this convex hull fn with something better
```

Comment:
Still a valid TODO?

Reply:
It is a valid todo, but this convex hull function works so far and isn't the bottleneck, so I think it can stay for now...